libs/pageserver_api/src/models.rs

```diff
@@ -234,6 +234,28 @@ impl TenantConfigRequest {
     }
 }
 
+#[derive(Debug, Serialize, Deserialize)]
+pub struct TenantAttachRequest {
+    pub config: TenantAttachConfig,
+}
+
+/// Newtype to enforce deny_unknown_fields on TenantConfig for
+/// its usage inside `TenantAttachRequest`.
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(deny_unknown_fields)]
+pub struct TenantAttachConfig {
+    #[serde(flatten)]
+    allowing_unknown_fields: TenantConfig,
+}
+
+impl std::ops::Deref for TenantAttachConfig {
+    type Target = TenantConfig;
+
+    fn deref(&self) -> &Self::Target {
+        &self.allowing_unknown_fields
+    }
+}
+
```

Comment:
[For option D in this comment](https://github.com/neondatabase/cloud/issues/5092#issuecomment-1554672222), I'd like to have the not-flattened `config` field, so that we can have another `old_config` field in the future.

Reply:
Ok, so, you think as well we should keep it like it is here, i.e., not-flattened?

Reply:
If I'm looking at the right thing here the flattened version is used. 

Yeah, I think the version without flatten should be the easiest to work with

Reply:
> If I'm looking at the right thing here the flattened version is used.

To avoid confusion, I'll re-iterate what the current tip of the branch does.
It will accept:

```
{
  "config": {
    "gc_period": "2h"
  }
}
```

it will not accept

```
{
  "gc_period": "2h"
}
```

If you think that's good (I think it is), please resolve this conversation.

Reply:
Yep, correct, thanks!